osf/management/commands/rebrand_preprint_provider.py

```diff
@@ -0,0 +1,117 @@
+"""
+This script is destructive and should be used with caution.
+Before use, a new provider must already be created with correct subjects and assets.
+
+A change to the nginx config, redirecting "/preprints/<src_id>/*" to "/preprints/<dst_id>/*"
+must go out when this is ran.
+"""
+from __future__ import unicode_literals
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Count
+import progressbar
+
+from scripts import utils as script_utils
+from osf.management.commands.reindex_preprint_provider import reindex_provider
+from osf.models import PreprintProvider
+from osf.utils.migrations import disable_auto_now_fields
+
+logger = logging.getLogger(__name__)
+
+def rebrand_provider(src_id, dst_id):
+    assert src_id != 'osf', 'Cannot rebrand OSF Preprints'
+    src_prov = PreprintProvider.load(src_id)
+    dst_prov = PreprintProvider.load(dst_id)
+    assert src_prov, 'Unable to find provider {}'.format(src_id)
+    assert dst_prov, 'Unable to find provider {}'.format(dst_id)
+    assert set(src_prov.subjects.values_list('text', flat=True)) == set(dst_prov.subjects.values_list('text', flat=True)), 'Provider subjects do not match'
```

Comment:
Should we also check for subjects_acceptable?

Reply:
Those aren't real.